src/test/java/dev/blaauwendraad/masker/json/path/JsonPathParserTest.java

```diff
@@ -40,24 +59,53 @@ private static Stream<Arguments> legalJsonPathLiterals() {
                 Arguments.of("$.a.b.c.", new JsonPath(new String[]{"$", "a", "b", "c"})),
                 Arguments.of("$[a].b.[c]", new JsonPath(new String[]{"$", "a", "b", "c"})),
                 Arguments.of("$.a.[b][c]", new JsonPath(new String[]{"$", "a", "b", "c"})),
-                Arguments.of("$.a[0].[b][c]", new JsonPath(new String[]{"$", "a", "0", "b", "c"})),
                 Arguments.of("$[][][]", new JsonPath(new String[]{"$", "", "", ""})),
-                Arguments.of("$[a][b][5].c", new JsonPath(new String[]{"$", "a", "b", "5", "c"})),
-                Arguments.of("$.a[0].b[1].c[2]", new JsonPath(new String[]{"$", "a", "0", "b", "1", "c", "2"}))
+                Arguments.of("$.a.*.c", new JsonPath(new String[]{"$", "a", "*", "c"})),
+                Arguments.of("$.*.b.*", new JsonPath(new String[]{"$", "*", "b", "*"})),
+                Arguments.of("$[*].*.*", new JsonPath(new String[]{"$", "*", "*", "*"}))
```

Comment:
Have you considered baning these combinations as well? (Basically equivalent to `$.*`)

Reply:
`$.a` and `$.a.*` would be equivalent, but cases with multiple leading `*` is trickier. 
`$.*.*.*` would query `c` only:
```
{
 "a": {
   "b": {
      "c": "target"
    },
   "d": "not target"
 },
 "f": "not target"
}
```

Reply:
Ah, yes. You are correct